### PR TITLE
update ray_version to 2.7.0

### PR DESCRIPTION
**Pull request by GeneDer**
## Why are these changes needed?

Ran the command to update ray version to 2.7.0

## Related issue number

Closes https://www.notion.so/How-to-release-Ray-go-release-ray-28e4a2347577471288bbd7d64001fa88#82b68d7c4b384f63a850419059e03272

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tests/test_runtime_env.py [hunk 1]
```diff
@@ -43,7 +43,7 @@
 def test_get_wheel_filename():
     """Test the code that generates the filenames of the `latest` wheels."""
     # NOTE: These should not be changed for releases.
-    ray_version = "3.0.0.dev0"
+    ray_version = "2.7.0"
```
Comment (aslonnie) [inline python/ray/tests/test_runtime_env.py:46]: these are failing the unit tests.

the note says that these should not be changed for releases.

File: java/api/pom_template.xml [hunk 2]
```diff
@@ -6,7 +6,7 @@
   <parent>
     <groupId>io.ray</groupId>
     <artifactId>ray-superpom</artifactId>
-    <version>2.0.0-SNAPSHOT</version>
+    <version>2.5.0</version>
```
Comment (aslonnie) [inline java/api/pom_template.xml:9]: why are these changed to 2.5.0 ? why 2.5.0?

Comment (GeneDer) [reply java/api/pom_template.xml:9]: ouch I must forget to change the version in the second command.

File: python/ray/tests/test_runtime_env.py (hunk 1 above)
Comment (GeneDer) [reply python/ray/tests/test_runtime_env.py:46]: Will fix it

**Comment by GeneDer** [comment]
raised new PR to fix it https://github.com/ray-project/ray/pull/38942
